Feature/games

sebastianturbiarz opened the pull request:
This closes #10, closes #7 , closes #19 , closes #6

sebastianturbiarz committed:
Create OurGames

sebastianturbiarz committed:
Add OurGames, opacity button background,

sebastianturbiarz committed:
Prepare to merge

sebastianturbiarz committed:
Prepare to merge

sebastianturbiarz committed:
Fixed container width

bibixx commented on frontend/components/Layout/Layout.jsx:
```diff
@@ -1,15 +1,27 @@
+Layout.propTypes = {
+  hasBear: PropTypes.bool,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
```
Why do you allow array of nodes? There is no need for thath

bibixx commented on frontend/components/OurGames/styles.js:
```diff
@@ -0,0 +1,14 @@
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  max-width: 1200px;
```
It'd be better to use `rems` here

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -0,0 +1,27 @@
+    <StyledPicture src={picture} />
+    <StrongP>{name}</StrongP>
+    <SubText>{subText}</SubText>
+    <StyledButton>Play now</StyledButton>
```
Remember that this will be a link

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -0,0 +1,27 @@
+
+const Game = ({ picture, name, subText }) => (
+  <StyledDiv>
+    <StyledPicture src={picture} />
```
Remember to **ALWAYS** add an alt tag. If the image doesn't give any value for the user you can simply add an empty alt (i.e. `alt=""`) but you should add it as if it is not present screen readers will read full link of the image and it won't really be meaningful

bibixx commented on frontend/components/OurGames/OurGames.jsx:
```diff
@@ -0,0 +1,50 @@
+  <StyledSection>
+    <Heading1>our games</Heading1>
+    <GamesContainer>
+      <Game
```
Maybe let's not hard code all of those as specific components, but rather let's create an array for now so that connecting it with api will be easier

bibixx commented on frontend/components/elements/Game/styles.js:
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+export const StyledDiv = styled('div')`
```
Styled div? Not really descriptive

bibixx commented on frontend/components/elements/Game/styles.js:
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+export const StyledDiv = styled('div')`
+  padding: 1rem 1rem;
```
You could have done
```suggestion
  padding: 1rem;
```

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -0,0 +1,27 @@
+import { StyledDiv, StyledPicture } from './styles';
+
+const Game = ({ picture, name, subText }) => (
+  <StyledDiv>
```
Probably every game should be a link leading to the specific game page

bibixx commented on frontend/components/OurGames/styles.js:
```diff
@@ -0,0 +1,14 @@
+  text-align: center;
+`;
+
+export const GamesContainer = styled('div')`
```
Probably it would be more semantic to make it a list (`ul` and `li`s)

sebastianturbiarz committed:
Add API, PropTypes, Link

Fixed after CR

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -0,0 +1,32 @@
+  picture, name, subText, playLink,
+}) => (
+  <GameContainer>
+    <StyledPicture src={picture} alt={name} />
```
I specifically said to give this an empty alt tag as it doesn't give any value for user

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -0,0 +1,32 @@
+    <StyledPicture src={picture} alt={name} />
+    <StrongP>{name}</StrongP>
+    <SubText>{subText}</SubText>
+    {/* <Link href="https://google.com"><StyledLink>Play now</StyledLink></Link> */}
```
Commented out code. Pls fix it

bibixx commented on frontend/components/elements/Button/Button.jsx:
```diff
@@ -0,0 +1,65 @@
+  borderColor: '255, 255, 255',
+};
+
+const StyledButton = styled('a')`
```
I would oppose to naming `a` tag a `StyledButton` it introduces unnecessary confusion

bibixx commented on frontend/.eslintrc.js:
```diff
@@ -6,7 +6,11 @@ module.exports = {
-      }
+      },
+      env: {
+        jest: true,
```
What is the need of adding `jest` env?

bibixx replied:
My bad I said to include it. Pls remove this one 😉

bibixx commented on frontend/components/Header/Header.jsx:
```diff
@@ -10,9 +10,9 @@ import {
+      <StyledLogo src={headerText} alt="Bodging Bear" />
     </LogoContainer>
-    <StyledBanner src={hasBear ? `${bannerBear}` : `${bannerNoBear}`} />
+    <StyledBanner src={hasBear ? `${bannerBear}` : `${bannerNoBear}`} alt="banner" />
```
I said to make `alt` empty

bibixx commented on frontend/components/elements/Contact/Contact.jsx:
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+const Contact = () => (
+  <h2>afasd</h2>
```
Maybe giving an actual text?

bibixx commented on frontend/components/elements/Button/Button.jsx:
```diff
@@ -0,0 +1,65 @@
+
+const StyledLink = ({ href, text }) => (
+  <Link prefetch href={href} passHref>
+    <StyledButton>{text}</StyledButton>
```
I think it would be more logical to use `children` instead of `text`

bibixx commented on frontend/pages/_document.jsx:
```diff
@@ -26,4 +28,16 @@ export default class MyDocument extends Document {
+
+  render() {
+    return (
+      <Html lang="eng">
```
It should be `en` instead of `eng`

bibixx commented on frontend/pages/index.jsx:
```diff
@@ -1,12 +1,44 @@
 );
 
+Index.getInitialProps = async function () {
+  const res = await fetch('http://localhost:3000/api/projects/');
```
We shouldn't make requests to certain urls as on production we won't have access to localhost and instead it will be hosted on a different domain

```suggestion
  const res = await fetch('/api/projects/');
```

sebastianturbiarz committed:
Fix after CR, 

Button, Alt's, API call

sebastianturbiarz committed:
Fix layout children, fix API Call

bibixx committed:
Fix 500 error

bibixx committed:
Add new field on backend, fix baseurl on fe

bibixx committed:
Fix index fetch on frontend

bibixx commented on frontend/components/Header/Header.jsx:
```diff
@@ -10,9 +10,9 @@ import {
   <StyledHeader>
     <LogoContainer>
-      <StyledLogo src={headerText} alt="Bodging Bear" />
+      <StyledLogo src={headerText} alt="" />
```
`Bodging Bear` in hear is ok

bibixx commented on frontend/components/elements/Game/Game.jsx:
```diff
@@ -8,11 +8,10 @@ const Game = ({
     <SubText>{subText}</SubText>
-    {/* <Link href="https://google.com"><StyledLink>Play now</StyledLink></Link> */}
-    <StyledLink href={playLink} text="Play now" />
+    <StyledLink href={playLink}>Play Now</StyledLink>
```
I just thought about it. Why are you using a next link to redirect to separate page? 🤔

bibixx committed:
Remove unnecessary file

sebastianturbiarz committed:
Fix text width in game description.

sebastianturbiarz committed:
Add dynamic routing for Game

sebastianturbiarz committed:
Fix after lint

Delete   "rules": {
    "sh-waqar/declaration-use-variable": "/color/"
  } in stylelint = error color: inherit;

sebastianturbiarz committed:
add component aboutGame
